atc/builds/planner.go

```diff
@@ -95,13 +95,14 @@ func (visitor *planVisitor) VisitGet(step *atc.GetStep) error {
 	visitor.plan = visitor.planFactory.NewPlan(atc.GetPlan{
 		Name: step.Name,
 
-		Type:     resource.Type,
-		Resource: resourceName,
-		Source:   resource.Source,
-		Params:   step.Params,
-		Version:  &version,
-		Tags:     step.Tags,
-		Timeout:  step.Timeout,
+		Type:                 resource.Type,
+		Resource:             resourceName,
+		Source:               resource.Source,
+		Params:               step.Params,
+		Version:              &version,
+		Tags:                 step.Tags,
+		Timeout:              step.Timeout,
+		ExposeBuildCreatedBy: resource.ExposeBuildCreatedBy,
```

Comment:
Can we avoid adding it to the `atc.GetPlan`? It seems like it should only be needed for `put`. 🤔 Passing it to `get` feels strange because it could technically end up caching stuff based on whoever triggered the build that generated the cache. I don't think it'll be useful to any resources anyway.

Reply:
@vito  Remember there are resource types to fetch build metadata, I wrote one for our internal usages, there are other community ones. For such resource type, `put` generates `build_id: xxx` as version, then implied `get` saves build metadata into files, so `get` needs it also.

I don't think cache is a problem. Because if a resource type doesn't store build metadata into files, then there is no cache problem there. If a resource type, like build metadata resource, BUILD_ID, BUILD_NAME may have same cache problem, so that it must use `put` to avoid cache problem.

I you have strong opinion to remove it from `get`, I'm fine. But in that way, build metadata resource's `put` will need to use include "create-by" in version, like `{build_id: 1234, created_by: someone}`, because version is the only way `put` can pass data to implied `get`.

WDYT?

Reply:
That's a good point, but as far as the original request goes it sounded like this was only necessary for notifications, which would use the information directly in the `put` step. It seems risky to extend this to support patterns beyond that as they might become deprecated when proper support for notifications is introduced.

Reply:
I'm fine as that won't block our requirements. I'll remove the exposing from `get`.